src/ui.rs

```diff
@@ -42,17 +46,24 @@ pub fn button(
                         "bg-gray-100 shadow-gray-200/10 hover:shadow-gray-200/20 text-gray-900"
                     }
                 },
-                class,
+                &props.class,
             ]
                 .join(" "),
-            onclick: onclick,
-            {text}
+            onclick: move |evt| onclick.clone().call(evt),
```

Comment:
Not sure whether this clone is needed:

```suggestion
            onclick: move |evt| onclick.call(evt),
```